File: angularapp/src/app/interfaces/order-model.ts
export interface OrderModel {
  id: number;
  userEmail: string;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  deliveryAddress: string;
  deliveryName: string;
  paymentName: string;
  status: string;
  orderDate: Date;
  isPaymentSuccessful: boolean;
  total: number;
  orderItems: OrderItemModel[];
}

export interface OrderItemModel {
  name: string;
price: number;
  quantity: number;
}

File: webapi/BLL/Models/OrderModel.cs
namespace webapi.BLL.Models
{
    public class OrderModel
    {
        public int Id { get; set; }
        public string UserEmail { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string PhoneNumber { get; set; }
        public string DeliveryAddress { get; set; }
        public string DeliveryName { get; set; }
        public string PaymentName { get; set; }
        public string Status { get; set; }
        public DateTimeOffset OrderDate { get; set; }
        public bool IsPaymentSuccessfull { get; set; }
        public decimal Total { get; set; }
        public IEnumerable<OrderItemModel> OrderItems { get; set; }
    }
}
